nx/lib/nx.ex

```diff
@@ -10987,6 +10987,69 @@ defmodule Nx do
     end)
   end
 
+  @doc """
+  Calculate the n-th discrete difference along the given axis.
+
+  The first difference is given by `out[i] = a[i+1] - a[i]` along the given axis,
+  higher differences are calculated by using `diff` recursively.
+
+  ## Options
+
+    * `:n` - the number of times to perform the difference. Defaults to `1`
+    * `:axis` - the axis to perform the difference along. Defaults to `-1`
+
+  ## Examples
+
+      iex> Nx.diff(Nx.tensor([1, 2, 4, 7, 0]))
+      #Nx.Tensor<
+        s64[4]
+        [1, 2, 3, -7]
+      >
+
+      iex> Nx.diff(Nx.tensor([1, 2, 4, 7, 0]), n: 2)
+      #Nx.Tensor<
+        s64[3]
+        [1, 1, -10]
+      >
+
+      iex> Nx.diff(Nx.tensor([[1, 3, 6, 10], [0, 5, 6, 8]]))
+      #Nx.Tensor<
+        s64[2][3]
+        [
+          [2, 3, 4],
+          [5, 1, 2]
+        ]
+      >
+
+      iex> Nx.diff(Nx.tensor([[1, 3, 6, 10], [0, 5, 6, 8]]), axis: 0)
+      #Nx.Tensor<
+        s64[1][4]
+        [
+          [-1, 2, 0, -2]
+        ]
+      >
+  """
+  @doc type: :cumulative
```

Comment:
I have no idea if this is the right category

Reply:
Given what we have, maybe window? Give it can be implemented as a convolution as well?

Reply:
Hmm, the 2D analogy isn't much intuitive for me.